plan/exhaust_physical_plans.go

```diff
@@ -421,12 +421,94 @@ func (p *LogicalJoin) getIndexJoinByOuterIdx(prop *requiredProp, outerIdx int) [
 		}
 	}
 	if bestIndexInfo != nil {
-		innerPlan := x.forceToIndexScan(bestIndexInfo, remainedOfBest)
+		innerPlan := p.constructInnerIndexScan(x, bestIndexInfo, remainedOfBest, outerJoinKeys)
 		return p.constructIndexJoin(prop, innerJoinKeys, outerJoinKeys, outerIdx, innerPlan, rangesOfBest, keyOff2IdxOff)
 	}
 	return nil
 }
 
+// constructInnerTableScan is specially used to construct the inner plan for PhysicalIndexJoin.
+func (p *LogicalJoin) constructInnerTableScan(ds *DataSource, pk *expression.Column, outerJoinKeys []*expression.Column) PhysicalPlan {
+	var ranges []*ranger.Range
+	if pk != nil {
+		ranges = ranger.FullIntRange(mysql.HasUnsignedFlag(pk.RetType.Flag))
+	} else {
+		ranges = ranger.FullIntRange(false)
+	}
+	ts := PhysicalTableScan{
+		Table:           ds.tableInfo,
+		Columns:         ds.Columns,
+		TableAsName:     ds.TableAsName,
+		DBName:          ds.DBName,
+		filterCondition: ds.pushedDownConds,
+		Ranges:          ranges,
+		rangeDecidedBy:  outerJoinKeys,
+	}.init(ds.ctx)
+	ts.SetSchema(ds.schema)
+
+	var rowCount float64
+	pkHist, ok := ds.statisticTable.Columns[pk.ID]
+	if ok && !ds.statisticTable.Pseudo {
+		rowCount = pkHist.AvgCountPerValue(ds.statisticTable.Count)
+	} else {
+		rowCount = ds.statisticTable.PseudoAvgCountPerCount()
+	}
+
+	ts.stats = newSimpleStats(rowCount)
+	ts.stats.usePseudoStats = ds.statisticTable.Pseudo
+
+	copTask := &copTask{
+		tablePlan:         ts,
+		indexPlanFinished: true,
+	}
+	ts.addPushedDownSelection(copTask, ds.stats)
+	t := finishCopTask(ds.ctx, copTask)
+	return t.plan()
+}
+
+// constructInnerIndexScan is specially used to construct the inner plan for PhysicalIndexJoin.
+func (p *LogicalJoin) constructInnerIndexScan(ds *DataSource, idx *model.IndexInfo, remainedConds []expression.Expression, outerJoinKeys []*expression.Column) PhysicalPlan {
+	is := PhysicalIndexScan{
+		Table:            ds.tableInfo,
+		TableAsName:      ds.TableAsName,
+		DBName:           ds.DBName,
+		Columns:          ds.Columns,
+		Index:            idx,
+		dataSourceSchema: ds.schema,
+		KeepOrder:        false,
+		Ranges:           ranger.FullRange(),
+		rangeDecidedBy:   outerJoinKeys,
+	}.init(ds.ctx)
+	is.filterCondition = remainedConds
+
+	var rowCount float64
+	idxHist, ok := ds.statisticTable.Indices[idx.ID]
+	if ok && !ds.statisticTable.Pseudo {
+		rowCount = idxHist.AvgCountPerValue(ds.statisticTable.Count)
+	} else {
+		rowCount = ds.statisticTable.PseudoAvgCountPerCount()
```

Comment:
The method name should be `PseudoAvgCountPerValue`?

Reply:
yes, already changed.